File: ConsoleApplication/LinkedList/Reverse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication1.LinkedList
{
    partial class LinkedList
    {
        public Node Reverse(Node head)
        {
            Node prev = null;
            Node cur = head;
            Node next = null;

            while (cur != null)
            {
                next = cur.Next;
                cur.Next = prev;
                prev = cur;
                cur = next;
            }

            return prev;
        }

    }
}

File: Tasks/src/linkedLists/Reverse.kt
package linkedLists

class Reverse {
    fun getRerversedHead() {

        var linkedList = LinkedList.getExample()
        var head = linkedList.head

        val newHead = reverse(head)
        println(newHead?.value)
    }

    private fun reverse(head: LinkedListNode?): LinkedListNode? {

        var prev: LinkedListNode? = null
        var cur: LinkedListNode? = head
        var next: LinkedListNode? = null

        while(cur != null) {
            next = cur.next
            cur.next = prev
            prev = cur
            cur = next
        }

        return  prev
    }
}